core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestCounterBasedAnonymizer.java

```diff
@@ -0,0 +1,120 @@
+import static io.trino.sql.tree.ComparisonExpression.Operator.LESS_THAN;
+import static io.trino.sql.tree.LogicalExpression.Operator.AND;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestCounterBasedAnonymizer
+{
+    @Test
+    public void testTimestampWithTimeZoneValueAnonymization()
+    {
+        CounterBasedAnonymizer anonymizer = new CounterBasedAnonymizer();
+        assertThat(anonymizer.anonymize(TIMESTAMP_TZ_MILLIS, "2012-10-30 18:00:00.000 America/Los_Angeles"))
+                .isEqualTo("timestamp_3_with_time_zone_value_1");
+    }
+
+    @Test
+    public void testSymbolReferenceAnonymization()
+    {
+        LogicalExpression expression = new LogicalExpression(AND, ImmutableList.of(
+                new ComparisonExpression(GREATER_THAN, new SymbolReference("a"), new LongLiteral("1")),
+                new ComparisonExpression(LESS_THAN, new SymbolReference("b"), new LongLiteral("2")),
+                new ComparisonExpression(EQUAL, new SymbolReference("c"), new LongLiteral("3"))));
+        CounterBasedAnonymizer anonymizer = new CounterBasedAnonymizer();
+        assertThat(anonymizer.anonymize(expression))
+                .isEqualTo("((\"symbol_1\" > 'long_literal_1') AND (\"symbol_2\" < 'long_literal_2') AND (\"symbol_3\" = 'long_literal_3'))");
+    }
+
+    @Test(dataProvider = "literals")
+    public void testLiteralAnonymization(Literal actual, String expected)
+    {
+        CounterBasedAnonymizer anonymizer = new CounterBasedAnonymizer();
+        assertThat(anonymizer.anonymize(actual)).isEqualTo(expected);
+    }
+
+    @DataProvider
+    public static Object[][] literals()
+    {
+        return new Object[][] {
+                {new BinaryLiteral("DEF321"), "'binary_literal_1'"},
+                {new StringLiteral("abc"), "'string_literal_1'"},
+                {new GenericLiteral("bigint", "1"), "'bigint_literal_1'"},
+                {new CharLiteral("a"), "'char_literal_1'"},
+                {new DecimalLiteral("123"), "'decimal_literal_1'"},
+                {new DoubleLiteral(String.valueOf(6554)), "'double_literal_1'"},
+                {new DoubleLiteral(String.valueOf(Double.MAX_VALUE)), "'double_literal_1'"},
+                {new LongLiteral(String.valueOf(6554)), "'long_literal_1'"},
+                {new LongLiteral(String.valueOf(Long.MAX_VALUE)), "'long_literal_1'"},
+                {new BooleanLiteral("true"), "true"},
+                {new TimeLiteral("03:04:05"), "'time_literal_1'"},
+                {new TimestampLiteral("2012-10-31 01:00 UTC"), "'timestamp_literal_1'"},
+                {new NullLiteral(), "null"},
+                {
+                    new IntervalLiteral("33", IntervalLiteral.Sign.POSITIVE, IntervalLiteral.IntervalField.DAY, Optional.empty()),
+                    "'interval_literal_1'"
+                },
+                {new TestLiteral(Optional.empty()), "literal_anonymized"},
+        };
+    }
+
+    private static class TestLiteral
```

Comment:
What's the purpose of this?

The AST is meant to be a closed hierarchy and not to be extended by third-parties. There's a lot of infrastructure that depends on knowing exactly what classes are part of the AST. Eventually, we'll update it to use Java 17's sealed types and this won't be possible to do at all and enforced at compile time.

Reply:
The idea here is to test the scenario that a new sub-class of Literal is added but `CounterBasedAnonymizer#anonymizeExpression` isn't updated to handle that new class. In this scenario, we should still anonymise the literal.
It's not strictly necessary though, let me know if you want this removed or handled differently.

Reply:
Yes, let’s remove it. It’s not a correct usage of the AST classes and will break in the future.

As long as the anonymizer fails if a new class is added (very unlikely), we’ll be able to catch that very quickly and add the relevant code, so we don’t even have to handle anonymization for the general case.

Reply:
I've removed this now and modified `CounterBasedAnonymizer#anonymizeExpression` to throw UnsupportedOperationException in case of un-handled Literal sub-class.